src/seml/database.py

```diff
@@ -19,7 +23,46 @@ def get_collection(collection_name, mongodb_config=None, suffix=None):
     return db[collection_name]
 
 
-def get_mongo_client(db_name, host, port, username, password, **kwargs):
+def retried_and_locked_ssh_port_forward(
+        retries_max=6, retries_delay=1, lock_file='~/seml_ssh.lock', lock_timeout=30, **ssh_config):
+    try:
+        from sshtunnel import SSHTunnelForwarder, BaseSSHTunnelForwarderError
+    except ImportError as e:
+        print('Opening ssh tunnel requires `sshtunnel` (e.g. `pip install sshtunnel`)')
+        raise e
+    try:
+        from filelock import Timeout, FileLock
+    except ImportError as e:
+        print('Opening ssh tunnel requires `filelock` (e.g. `pip install filelock`)')
```

Comment:
Use `logging.error` and fail gracefully with an `exit` instead of throwing the exception to the user.
Should we add them as (optional) dependencies to `pyproject.toml`? maybe such that we can do `pip install seml[ssh_forward]`.